app/models/manageiq/providers/regions.rb

```diff
@@ -0,0 +1,46 @@
+module ManageIQ::Providers
+  class Regions
+    class << self
+      def regions
+        from_file.merge(additional_regions).except(*disabled_regions)
+      end
+
+      def all
+        regions.values
+      end
+
+      def names
+        regions.keys
+      end
+
+      def find_by_name(name)
+        regions[name]
+      end
+
+      private
+
+      def from_file
+        # Memoize the regions file as this should not change at runtime
+        @from_file ||= YAML.load_file(regions_yml).each_value(&:freeze).freeze
```

Comment:
Minor, but it might be better to use transform_values (I'm surprised rubocop didn't complain about this [[ref]](https://docs.rubocop.org/rubocop/cops_style.html#stylehashtransformvalues), but probably because you aren't really supposed to mutate the value you are iterating with `each_value`)

```suggestion
        @from_file ||= YAML.load_file(regions_yml).transform_values(&:freeze).freeze
```